src/dict.c

```diff
@@ -1513,6 +1513,19 @@ size_t _dictGetStatsHt(char *buf, size_t bufsize, dict *d, int htidx) {
             "No stats available for empty dictionaries\n");
     }
 
+    if (!full) {
+        l += snprintf(buf+l,bufsize-l,
+            "Hash table %d stats (%s):\n"
+            " table size: %lu\n"
+            " number of elements: %lu\n",
+            htidx, (htidx == 0) ? "main hash table" : "rehashing target",
+            DICTHT_SIZE(d->ht_size_exp[htidx]), d->ht_used[htidx]);
+
+        /* Unlike snprintf(), return the number of characters actually written. */
+        if (bufsize) buf[bufsize-1] = '\0';
```

Comment:
No point in this check, if we're concerned we'll get `bufsize==0`, we need to bail out before even calling `snprintf`.

Reply:
i really don't know what he was aiming for, it seems unnecessary, but i suppose it's safer to keep it than drop it.